sympy/physics/control/lti.py

```diff
@@ -1422,3 +1424,326 @@ def _eval_rewrite_as_TransferFunction(self, num, den, **kwargs):
+    >>> tfm_3[:, 0]  # gives the first column
+    TransferFunctionMatrix(((TransferFunction(5, s, s),), (TransferFunction(5, s*(s**2 + 2), s),)))
+    >>> pprint(_, use_unicode=False)
+    [    5     ]
+    [    -     ]
+    [    s     ]
+    [          ]
+    [    5     ]
+    [----------]
+    [  / 2    \]
+    [s*\s  + 2/]
+    >>> tfm_3[0, :]  # gives the first row
+    TransferFunctionMatrix(((TransferFunction(5, s, s), TransferFunction(5*s, s**2 + 2, s)),))
+    >>> pprint(_, use_unicode=False)
+    [5   5*s  ]
+    [-  ------]
+    [s   2    ]
+    [   s  + 2]
+
+    To negate a transfer function matrix, ``-`` operator can be prepended:
+
+    >>> tfm_4 = TransferFunctionMatrix([[tf_2], [-tf_1], [tf_3]])
+    >>> -tfm_4
+    TransferFunctionMatrix(((TransferFunction(-p**4 + 3*p - 2, p + s, s),), (TransferFunction(a + s, s**2 + s + 1, s),), (TransferFunction(-3, s + 2, s),)))
+    >>> tfm_5 = TransferFunctionMatrix([[tf_1, tf_2], [tf_3, -tf_1]])
+    >>> -tfm_5
+    TransferFunctionMatrix(((TransferFunction(-a - s, s**2 + s + 1, s), TransferFunction(-p**4 + 3*p - 2, p + s, s)), (TransferFunction(-3, s + 2, s), TransferFunction(a + s, s**2 + s + 1, s))))
+
+    ``subs()`` returns an ImmutableDenseMatrix object with the value substituted in the expression. This will not
+    mutate your original ``TransferFunctionMatrix``.
+
+    >>> from sympy import I
+    >>> tfm_3.subs({s: I})  #  substituting imaginary i (iota) to s
+    Matrix([
+    [-5*I, 5*I],
+    [-5*I, 5/1]])
+
+    See Also
+    ========
+
+    TransferFunction, Series, Parallel, Feedback
+
+    """
+    def __new__(cls, arg):
+
+        expr_mat_arg = []
+        var = arg[0][0].var
+        for row in range(len(arg)):
+            temp = []
+            for col in range(len(arg[row])):
+                if not isinstance(arg[row][col], (TransferFunction, Series, Parallel)):
+                    raise TypeError("Incompatible type found as the element of TransferFunctionMatrix")
+
+                if var != arg[row][col].var:
+                    raise ValueError("Conflicting value(s) found for `var`. All TransferFunction instances in "
+                            "TransferFunctionMatrix should use the same complex variable in Laplace domain.")
+
+                if isinstance(arg[row][col], TransferFunction):
+                    temp.append(arg[row][col].to_expr())
+                else:
+                    temp.append(arg[row][col].doit().to_expr())
+            expr_mat_arg.append(temp)
+
+
+        if isinstance(arg, (list, Tuple)):
+            # Making nested Tuple (sympy.core.containers.Tuple) from nested list or nested python tuple
+            arg = Tuple(*(Tuple(*r, sympify=False) for r in arg), sympify=False)
+
+        obj = super(TransferFunctionMatrix, cls).__new__(cls, arg)
+        obj._expr_mat = ImmutableMatrix(expr_mat_arg)
+        obj._var = arg[0][0].var
+        obj._num_outputs, obj._num_inputs = obj._expr_mat.shape
+        return obj
```

Comment:
You should avoid storing additional objects on the instance if possible. Ideally all of these would be derived through property methods from the args. Does this class satisfy `obj == obj.func(*obj.args)`?

Reply:
Yes. It does satisfy the `obj.func(*obj.args)` condition. In the old PR by @namannimmo10 it was failing and had a `@XFAIL` decorator, which I've removed after it was passing. You can check the changes in test_args.py file.